src/dparse/lexer.d

```diff
@@ -2001,9 +1968,297 @@ unittest
             assert(pair[1] == app.data, "[[" ~ pair[0] ~ "]] => [[" ~ app.data ~ "]]");
         }
     }
+    foreach (pair; np)
+    {
+        auto app = appender!string();
+        unDecorateComment(pair[0], app);
+        assert(pair[1] == app.data, "[[" ~ pair[0] ~ "]] => [[" ~ app.data ~ "]]");
+    }
     stderr.writeln("Unittest for unDecorateComment passed.");
 }
 
+/** Gives a line per line view on DDOC comments of type `/++` and `/**` which
+ * makes easier to remove the decoration and in an almost 100% nogc way.
+ * drawback: to remain fast and nogc only up to MaxLines lines are handled
+ */
+private struct MultiLineCommentHelper(CharType, ushort MaxLines)
+{
+private:
+
+    char[][MaxLines] lines;
```

Comment:
I don't know. It takes 8192 bytes of stack space under x86_64, which doesn't sound too crazy. 
IIRC you already encountered a problem once and you had to increase the fiber size.

over 512 lines of documentation is certainly not made for the little hint window you got after a mouse motion in the editor but rather to gen to doc... and even. What's the biggest module paragraph in phobos ?

Reply:
std.getopt.getopt function ddoc is made of 370 lines. That's the biggest i have in mind.

Reply:
this would also cover module documentation right? These can grow a lot larger than the functions if they summarize a lot and have a long example.

[nanovega](https://github.com/adamdruppe/arsd/blob/master/nanovega.d) and [simpledisplay](https://github.com/adamdruppe/arsd/blob/master/simpledisplay.d) have around 430 lines of module documentation which could grow even further if they add another example or two in there.

Reply:
Okay, I've abandoned the static array.